code/modules/economy/economy_machinery/atm.dm

```diff
@@ -0,0 +1,304 @@
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "atm"
+	anchored = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 10
+	density = FALSE
+	restricted_bypass = TRUE
+
+	///Current money account the ATM is accessing
+	var/datum/money_account/authenticated_account
+	///ID Card that is currently inserted into the ATM
+	var/obj/item/card/held_card
+
+	///UI screen the ATM is on currently
+	var/view_screen = ATM_SCREEN_DEFAULT
+	///cooldown inbetween printing balance statements n stuff
+	var/print_cooldown = 0
+	///the time when the lockout on the ATM is lifted
+	var/lockout_time = 0
+	///failed login attempts counter, used for locking out the atm
+	var/login_attempts = 0
+
+/obj/machinery/economy/atm/Initialize(mapload)
+	. = ..()
+	update_icon()
+
+/obj/machinery/economy/atm/update_icon_state()
+	. = ..()
+	if(stat & NOPOWER)
+		icon_state = "atm_off"
+	else
+		icon_state = "atm"
+
+/obj/machinery/economy/atm/update_overlays()
+	. = ..()
+	underlays.Cut()
+
+	if(stat & NOPOWER)
+		return
+
+	underlays += emissive_appearance(icon, "atm_lightmask")
+
+/obj/machinery/economy/atm/power_change()
+	..()
+	if(stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(1, LIGHTING_MINIMUM_POWER)
+	update_icon()
+
+/obj/machinery/economy/atm/process()
+	if(stat & NOPOWER)
+		return
+
+/obj/machinery/economy/atm/attack_hand(mob/user)
+	if(..())
+		return TRUE
+	if(issilicon(user))
+		to_chat(user, "<span class='warning'>Artificial unit recognized. Artificial units do not currently receive monetary compensation, as per Nanotrasen regulation #1005.</span>")
+		return
+	if(!account_database)
+		reconnect_database()
+
+	ui_interact(user)
+
+/obj/machinery/economy/atm/attack_ghost(mob/user)
+	ui_interact(user)
+
+/obj/machinery/economy/atm/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/card))
+		if(powered())
+			handle_id_insert(I, user)
+	else if(authenticated_account)
+		if(istype(I, /obj/item/stack/spacecash))
+			if(!powered())
+				return
+			insert_cash(I, user)
+
+	return ..()
+
+/obj/machinery/economy/atm/insert_cash(obj/item/stack/spacecash/cash_money, mob/user)
+	visible_message("<span class='info'>[user] inserts [cash_money] into [src].</span>")
+	cash_stored += cash_money.amount
+	account_database.credit_account(authenticated_account, cash_money.amount, "ATM Deposity", machine_id, FALSE)
+	cash_money.use(cash_money.amount)
+	return TRUE
+
+/obj/machinery/economy/atm/proc/handle_id_insert(obj/item/card/id, mob/user)
+	if(held_card)
+		return
+	user.drop_item()
+	id.forceMove(src)
+	held_card = id
+	RegisterSignal(held_card, COMSIG_PARENT_QDELETING, .proc/clear_held_card)
+	if(authenticated_account && held_card.associated_account_number != authenticated_account.account_number)
+		authenticated_account = null
+
+/obj/machinery/economy/atm/proc/eject_inserted_id(mob/user)
+	if(!held_card)
+		return
+	held_card.forceMove(loc)
+	UnregisterSignal(held_card, COMSIG_PARENT_QDELETING)
+	if(ishuman(user) && !user.get_active_hand())
```

Comment:
```suggestion
	if(ishuman(user)
```
Ended up reverting #19164 changes for the ATM.

Reply:
fixed